internal/search/query/types.go

```diff
@@ -102,16 +102,27 @@ func (q AndOrQuery) StringValue(field string) (value, negatedValue string) {
 
 func (q AndOrQuery) Values(field string) []*types.Value {
 	var values []*types.Value
-	VisitField(q.Query, field, func(value string, _, quoted bool) {
-		values = append(values, valueToTypedValue(field, value, quoted)...)
-	})
+	if field == "" || field == "content" {
+		VisitPattern(q.Query, func(value string, _, quoted bool) {
+			values = append(values, valueToTypedValue(field, value, quoted)...)
+		})
+	} else {
+		VisitField(q.Query, field, func(value string, _ bool) {
+			values = append(values, valueToTypedValue(field, value, false)...)
+		})
+	}
 	return values
 }
 
 func (q AndOrQuery) Fields() map[string][]*types.Value {
 	fields := make(map[string][]*types.Value)
-	VisitParameter(q.Query, func(field, value string, _, quoted bool) {
-		fields[field] = valueToTypedValue(field, value, quoted)
+	VisitPattern(q.Query, func(value string, _, quoted bool) {
+		// FIXME. NOTE: overrides. Seems not good, how is Fields() used?
+		// For existence? What about "content"?
+		fields[""] = valueToTypedValue("", value, quoted)
```

Comment:
Does this look right? (definitely should avoid giving meaning to empty string whenever possible)
```suggestion
		fields["content"] = valueToTypedValue("", value, quoted)
```

Or is it possible `fields["content"]` previously have value that we want to concatenate?

Reply:
Ah, thanks for catching this comment, I meant to address and remove it.

The issue here is that `Fields()` satisfies an existing interface of "values that our search code uses to figure out what to do". Here are [all the uses](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+%5C.Fields%5C%28%5C%29+lang:go&patternType=regexp#1). It's not clear how, what, or whether something is expected for `Fields()[FieldDefault]` AKA `Fields()[""]`. The empty string here is exactly how we currently store this information in `Fields()`, so I am just doing the same and trying to satisfy the interface (though, thanks to this PR, that will not be needed once we migrate away). After looking at everything, the safe thing to do is right now is to rely on `Values()` to supply the `Value` of fields, despite this doing extra work for visiting the query. I'll remove this interface later.